pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -1041,5 +1052,20 @@ private ExecutorService getInternalExecutor(Message<T> msg) {
         return executor;
     }
 
+    // If message consumer epoch is smaller than consumer epoch present that
+    // it has been sent to the client before the user calls redeliverUnacknowledgedMessages, this message is invalid.
+    // so we should release this message and receive again
+    protected boolean checkMessageConsumerEpochIsSmallerThanConsumer(MessageImpl<T> message) {
```

Comment:
For more safety, we should not filter out any message for a broker with an old version protocol version.